src/main/java/org/commonjava/util/sidecar/interceptor/MetricsHandlerInterceptor.java

```diff
@@ -35,6 +36,7 @@
 
 @Interceptor
 @MetricsHandler
+@RegisterForReflection
```

Comment:
Why should this Interceptor use this RegisterForReflection annotation here?

Reply:
It's for native compilation, do we want to remove everything related to that?

Reply:
Yes I know it's for native compilation. Question here is: is any reflection code used in this interceptor? Or is is a must for Quarkus framework to register all CDI interceptors with this annotation? If yes of one of these two questions, I think it is ok to add this annotation.

Reply:
It's must for any class without any constructor call, the annotation is only for compiler to not cut out these pieces of code, other reflected class don't need this.